R/restore.R

```diff
@@ -202,22 +202,19 @@ getSourceForPkgRecord <- function(pkgRecord,
       error = function(e) "internal"
     )
 
-    protocol <- if (identical(method, "internal"))
-      "http"
-    else
-      "https"
+    fmt <- "%s/repos/%s/%s/tarball/%s"
 
-    fmt <- "%s://api.github.com/repos/%s/%s/tarball/%s"
-    archiveUrl <- sprintf(fmt, protocol, pkgRecord$gh_username, pkgRecord$gh_repo, pkgRecord$gh_sha1)
+    archiveUrl <- sprintf(fmt, pkgRecord$remote_host, pkgRecord$gh_username, pkgRecord$gh_repo, pkgRecord$gh_sha1)
```

Comment:
The `remote_host` field was added with version v1.13.5 of devtools, if I'm understanding correctly:

https://github.com/rstudio/packrat/blob/3eb9c1250ae4384012164a75282f6992c4aa972e/R/github.R#L6

That was quite a while ago, but there's a chance there's still older R installations that don't have this version installed. Can we check to see if `pkgRecord$remote_host` is `NULL` or empty, and fallback to the GitHub API URL if it's not available?